grapple/utils.py

```diff
@@ -9,39 +12,68 @@
 from .types.structures import BasePaginatedType, PaginationType
 
 
-def resolve_site(hostname):
+def resolve_site(
+    *,
+    id: Optional[int] = None,
+    hostname: Optional[str] = None,
+    hostname_filter_name: Optional[Literal["site", "hostname"]] = None,
+) -> Optional[Site]:
```

Comment:
I have to say I am not enamored with this. the if dance just doesn't sit well with me.
My preference here would be to split this into `resolve_site_by_id(id: int)` and `resolve_site_by_hostname(hostname: str, filter_name: Literal["site", "hostname"])` and then the logic inside is easier to follow, and we don't need to mix .get/DoesNotExist with MultipleObjectsReturned

Reply:
Good shout - because the method originally handled both I didn't consider changing it, though this is much simpler!